app/main/views/send.py

```diff
@@ -846,38 +837,85 @@ def send_notification(service_id, template_id):
+        values.append(v)
+
+    data = ",".join(keys)
+    vals = ",".join(values)
+    data = f"{data}\r\n{vals}"
+
+    filename = f"one-off:{current_user.name}:{uuid.uuid4()}.csv"
+    my_data = {"filename": filename, "template_id": template_id, "data": data}
+    upload_id = s3upload(service_id, my_data)
+    form = CsvUploadForm()
+    form.file.data = my_data
+    form.file.name = filename
+
+    check_message_output = check_messages(service_id, template_id, upload_id, 2)
+    if "You cannot send to" in check_message_output:
+        return check_messages(service_id, template_id, upload_id, 2)
+
+    job_api_client.create_job(
+        upload_id,
+        service_id,
+        scheduled_for="",
+        template_id=template_id,
+        original_file_name=filename,
+        notification_count=1,
+        valid="True",
     )
 
-    try:
-        noti = notification_api_client.send_notification(
-            service_id,
-            template_id=db_template["id"],
-            recipient=recipient,
-            personalisation=session["placeholders"],
-            sender_id=session.get("sender_id", None),
+    session.pop("recipient")
+    session.pop("placeholders")
+
+    # We have to wait for the job to run and create the notification in the database
+    time.sleep(0.1)
+    notis = notification_api_client.get_notifications_for_service(
+        service_id, job_id=upload_id, include_one_off=True
+    )
+    attempts = 0
+    while notis["total"] == 0 and attempts < 5:
+        notis = notification_api_client.get_notifications_for_service(
+            service_id, job_id=upload_id, include_one_off=True
         )
-    except HTTPError as exception:
-        current_app.logger.error(
-            'Service {} could not send notification: "{}"'.format(
-                current_service.id, exception.message
+        time.sleep(0.1)
+        attempts = attempts + 1
+    # TODO we are replacing the original 'one-off send' functionality with a job that
+    # we create on the fly.  The purpose for this is to ultimately remove the phone numbers
+    # from the db.  However, by running a job we no longer get error messages we used to get.
+    # If the user is in trial mode and trying to send to a phone number they are not allowed
+    # to send to, right now they will see that their job started and the only way they will
+    # know something went wrong, is to sit and watch the status sit as pending for 3 hours
+    # and ultimately switch to failed with no reason why.
+    #
+    # In future, we should block the user from sending to phone numbers they aren't allowed
+    # to send to.
+    # the way to do that would be to make this available to the front end:
+    #
+    # <api>/service/utils/service_allowed_to_send_to
+    #
+    # After that the UI should be making this call as part of the phone number validation
+    # A user in trial mode should not be able to 'send message' to a phone number they are not
+    # allowed to send to
```

Comment:
Noting here to make an issue out of this `TODO` once this is merged, thanks for flagging it!

Reply:
Ha!  I flagged it, but then I fixed it.  I have removed the TODO.  After the conversation I had with Tim where he encouraged me to make the new approach work exactly the same as the old approach, a relevant error message appears now.

Reply:
Oh, even better! :-)  Thank you!